pkg/controllers/dynakube/oneagent/daemonset/arguments.go

```diff
@@ -14,23 +14,14 @@ const argumentPrefix = "--"
 const customArgumentPriority = 2
 const defaultArgumentPriority = 1
 
-func (b *builder) arguments() ([]string, error) {
+func (b *builder) arguments() []string {
 	argMap := prioritymap.New(
 		prioritymap.WithSeparator(prioritymap.DefaultSeparator),
 		prioritymap.WithPriority(defaultArgumentPriority),
 		prioritymap.WithAllowDuplicates(),
 	)
 
-	isProxyAsEnvDeprecated, err := isProxyAsEnvVarDeprecated(b.dk.OneAgentVersion())
-	if err != nil {
-		return []string{}, err
-	}
-
-	if !isProxyAsEnvDeprecated {
-		// deprecated
-		b.appendProxyArg(argMap)
-	}
-
+	b.appendProxyArg(argMap)
```

Comment:
should we always set `--proxy=` even if proxy is set? That would mean it's configured via file and args at the same time for the oneagent.

Reply:
```golang
// deprecated
func (b *builder) appendProxyArg(argMap *prioritymap.Map) {
	if b.hasProxy() {
		argMap.Append(argumentPrefix+"set-proxy", "$(https_proxy)")
	}
	// if no proxy is set, we still have to set it as empty to clear proxy settings the OA might have cached
	argMap.Append(argumentPrefix+"set-proxy", "")
}
```
`--set-proxy=` to be correct. The information I got is that to definitely delete the container's configured proxy you have to unmount and also set the argument, if not, its deletion is not forced.

Reply:
do we need to enforce the deletion if there is already a different proxy set via file or could that cause issues? We disable proxy via args, but set it to a value via the file.

Reply:
Pinging @StefanHauth , he knows more details

Reply:
`--set-proxy=""` should only be set if no proxy is configured. If a proxy is configured it should be set via file (on agent versions that support it. 
I don't know if updating the file content is automatically considered by the agent. I'll check that out.

Reply:
Update: `--set-proxy` always takes precedence. So we have to make sure that we either use `--set-proxy=""`or mount the files. If files are updated, OneAgent will pick up the changes and use the new proxy config on restart.